test/tests/functional/pbs_equiv_classes.py

```diff
@@ -1606,3 +1606,190 @@ class instead of the class it was in when it was queued.
         self.server.deljob(jid3, wait=True)
 
         self.server.expect(JOB, {'job_state': 'R'}, id=jid5)
+
+    def test_multiple_job_preemption_order(self):
+        """
+        Test that when multiple jobs from same eqivalence class are
+        preempted in reverse order they were created in and they are placed
+        into the same equivalence class
+        """
+
+        # Create 1 vnode with 3 ncpus
+        a = {'resources_available.ncpus': 3}
+        self.server.create_vnodes('vnode', a, 1, self.mom, usenatvnode=True)
+
+        # Create expressq
+        a = {'queue_type': 'execution', 'started': 'true',
+             'enabled': 'true', 'priority': 150}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='expressq')
+
+        # Submit 3 jobs with delay of 1 sec
+        # Delay of 1 sec will preempt jid3 and then jid2.
+        a = {'Resource_List.ncpus': 1}
+        J = Job(TEST_USER, attrs=a)
+        jid1 = self.server.submit(J)
+
+        time.sleep(1)
+
+        J2 = Job(TEST_USER, attrs=a)
+        jid2 = self.server.submit(J2)
+
+        time.sleep(1)
+        J3 = Job(TEST_USER, attrs=a)
+        jid3 = self.server.submit(J3)
+
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid3)
+
+        # Preempt jid3 with expressq, check 1 equivalence class is created
+        a = {'Resource_List.ncpus': 1, 'queue': 'expressq'}
+        Je = Job(TEST_USER, attrs=a)
+        jid4 = self.server.submit(Je)
+
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid3)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid4)
+
+        self.scheduler.log_match("Number of job equivalence classes: 2",
+                                 max_attempts=10, starttime=self.t)
+
+        # Preempt jid2, check no new equivalence class is created
+        Je2 = Job(TEST_USER, attrs=a)
+        jid5 = self.server.submit(Je2)
+
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid2)
+        self.server.expect(JOB, {'job_state': 'S'}, id=jid3)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid4)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid5)
+
+        # Only One equivalence class for jid2 and jid3 is present since both
+        # suspended jobs are of same type and running on same vnode
+        t = int(time.time())
+        self.scheduler.log_match("Number of job equivalence classes: 2",
+                                 max_attempts=10, starttime=self.t)
```

Comment:
Looking at this again and test_preemption2, I would suggest extend this test to submit 2 more jobs under same equivalence class and deleting one of the running jobs. queued job will run and equivalence class will remain the same.